src/analyzer/plugins/analyzerkeyfinder.cpp

```diff
@@ -0,0 +1,123 @@
+#include "analyzer/plugins/analyzerkeyfinder.h"
+
+#include "analyzer/constants.h"
+#include "util/assert.h"
+#include "util/math.h"
+
+using mixxx::track::io::key::ChromaticKey;
+using mixxx::track::io::key::ChromaticKey_IsValid;
+
+namespace mixxx {
+
+AnalyzerKeyFinder::AnalyzerKeyFinder()
+        : m_pKeyFinder(nullptr),
+          m_pWorkspace(nullptr),
+          m_pAudioData(nullptr),
+          m_currentFrame(0),
+          m_previousKey(mixxx::track::io::key::INVALID) {
+}
+
+AnalyzerKeyFinder::~AnalyzerKeyFinder() {
+}
+
+bool AnalyzerKeyFinder::initialize(int samplerate) {
+    m_resultKeys.clear();
+    m_currentFrame = 0;
+    m_previousKey = mixxx::track::io::key::INVALID;
+
+    m_pKeyFinder = std::make_unique<KeyFinder::KeyFinder>();
+    m_pWorkspace = std::make_unique<KeyFinder::Workspace>();
+    m_pAudioData = std::make_unique<KeyFinder::AudioData>();
+    m_pAudioData->setFrameRate(samplerate);
+    m_pAudioData->setChannels(kAnalysisChannels);
+
+    return true;
+}
+
+bool AnalyzerKeyFinder::processSamples(const CSAMPLE* pIn, const int iLen) {
+    DEBUG_ASSERT(iLen % kAnalysisChannels == 0);
+    if (m_pAudioData->getSampleCount() == 0) {
+        m_pAudioData->addToSampleCount(iLen);
+    }
+
+    const size_t numInputFrames = iLen / kAnalysisChannels;
```

Comment:
Prefer to use consistent types within a context. Here we currently mix 3 different types for a single purpose: int / size_t / unsigned int. When using a single loop we don't need this variable anyway.